### PR TITLE
test/system: add test for mounting issue in the init container

### PR DESCRIPTION
**Pull request by chuanchang**
Test coverage for https://issues.redhat.com/browse/RHEL-14469

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: test/system/030-run.bats [hunk 1]
```diff
@@ -1382,4 +1382,19 @@ search               | $IMAGE           |
     run_podman container rm $cname
 }
 
+# the /run should not be mounted as 777 in the init container
+@test "podman run - the /run is mounted as 777 in the init container with crun runtime" {
+    runtime=$(podman_runtime)
+    cname=$(random_string)
+    init_img="registry.access.redhat.com/ubi8/ubi-init"
+
+    run_podman run --runtime $runtime --name $cname -d --rm --network none --cap-add all $init_img /usr/sbin/init
```
Comment (vrothberg) [inline test/system/030-run.bats:1391]: I "think" it should also work with `--systemd=always $IMAGE`.  This way we'd avoid pulling down an image.

@edsantiago WDYT?

**Review by vrothberg — COMMENTED** [review]
Thanks!

File: test/system/030-run.bats (hunk 1 above)
Comment (edsantiago) [reply test/system/030-run.bats:1391]: Yes, please try never to pull anything in tests. Other problems too, I'll send more feedback soon.

**Comment by edsantiago** [comment]
Suggestion:
```bash
# https://issues.redhat.com/browse/RHEL-14469
@test "podman run - /run must not be world-writable in systemd containers" {
    # FIXME FIXME! Please document why --network none and --cap-add all are needed here
    run_podman run --name mycontainer -d --network none --cap-add all $SYSTEMD_IMAGE /usr/sbin/init

    # runc has always been 755; crun < 1.11 was 777
    run_podman exec mycontainer stat -c '%a' /run
    assert "$output" = "755" "stat /run"

    run_podman rm -f -t0 mycontainer
}
```

File: test/system/030-run.bats (hunk 1 above)
Comment (chuanchang) [reply test/system/030-run.bats]: Done, thank you for your review!

File: test/system/030-run.bats [hunk 2]
```diff
@@ -1382,4 +1382,14 @@ search               | $IMAGE           |
     run_podman container rm $cname
 }
 
+# https://issues.redhat.com/browse/RHEL-14469
+@test "podman run - /run must not be world-writable in systemd containers" {
+    run_podman run -d --rm $SYSTEMD_IMAGE /usr/sbin/init
+    cid=$output
+    run_podman exec -it $cid stat -c '%a' /run
```
Comment (edsantiago) [inline test/system/030-run.bats:1389]: I put a comment there. I did so for a reason: if the test ever fails, and someone goes looking into the code, this comment will make it possible for them to understand what's going on. Without that comment, this is just magic.
```bash
    # runc has always been 755; crun < 1.11 was 777
```

File: test/system/030-run.bats [hunk 3]
```diff
@@ -1382,4 +1382,14 @@ search               | $IMAGE           |
...
+    assert "$output" =~ "755" "stat /run"
```
Comment (edsantiago) [inline test/system/030-run.bats:1390]: *equals*, not equal-tilde please.

**Review by edsantiago — CHANGES_REQUESTED** [review]
Almost!

File: test/system/030-run.bats (hunk 3 above)
Comment (chuanchang) [reply test/system/030-run.bats:1390]: Nope, the output of "stat -c '%a' /run" is '755/r' or '1777/r' on the RHEL 8.y.

File: test/system/030-run.bats (hunk 2 above)
Comment (chuanchang) [reply test/system/030-run.bats:1389]: Make sence.

File: test/system/030-run.bats (hunk 3 above)
Comment (edsantiago) [reply test/system/030-run.bats:1390]: Use `exec`, not `exec -it`. `-it` is almost always a bad idea.

File: test/system/030-run.bats (hunk 2 above)
Comment (chuanchang) [reply test/system/030-run.bats]: Done

File: test/system/030-run.bats (hunk 3 above)
Comment (chuanchang) [reply test/system/030-run.bats]: Done.

**Comment by martinpitt** [comment]
Quite obviously this PR doesn't break cockpit. The rawhide run got confused somehow:
```
cp: cannot stat '/home/admin/.local/share/containers/storage/overlay/metacopy-check3136313145': No such file or directory
```
This seems obscure enough to ignore here.

**Pull request merged**